File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/resources/MusicResource.java
package pize.tests.minecraft.game.resources;

import pize.audio.sound.Sound;
import pize.tests.minecraft.utils.log.Logger;

public class MusicResource extends Resource<Sound>{

    private Sound sound;
    private boolean loaded;

    public MusicResource(String location){
        super(location);
    }

    @Override
    public void loadResource(){
        try{
            sound = new Sound(getLocation());
            loaded = true;
        }catch(RuntimeException e){
            loaded = false;
            Logger.instance().warn(e.getLocalizedMessage());
        }
    }

    @Override
    public void reloadResource(){
        Sound oldSound = sound;
        loadResource();
        oldSound.dispose();
    }

    @Override
    public boolean isLoaded(){
        return loaded;
    }

    @Override
    public Sound getResource(){
        return sound;
    }

    @Override
    public void dispose(){
        sound.dispose();
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/resources/MusicResource.kt
package pize.tests.minecraft.game.resources

import pize.audio.sound.Sound
import pize.tests.minecraft.utils.log.Logger

class MusicResource(location: String) : Resource<Sound?>(location) {
    override var resource: Sound? = null
        private set
    override var isLoaded = false
        private set

    override fun loadResource() {
        try {
            resource = Sound(location)
            isLoaded = true
        } catch (e: RuntimeException) {
            isLoaded = false
            Logger.Companion.instance().warn(e.localizedMessage)
        }
    }

    override fun reloadResource() {
        val oldSound = resource
        loadResource()
        oldSound!!.dispose()
    }

    override fun dispose() {
        resource!!.dispose()
    }
}
